### ISSUE TITLE
Request: ignoreSingleValue for requirePaddingNewLinesInObjects

### ISSUE THREAD

**Issue by yaycmyk** [jscs-dev/node-jscs#1115]
Currently it only accepts `true`, but it would be nice to also accept `ignoreSingleValue`, similar to `requireTrailingComma` (thanks @hzoo).

Valid:
````javascript
// requirePaddingNewLinesInObjects: 'ignoreSingleValue'

var a = {b: 'c'};
var b = {
    b: 'c',
    c: 'd'
};
````

Invalid:
````javascript
// requirePaddingNewLinesInObjects: 'ignoreSingleValue'

var b = {b: 'c', c: 'd'};
````

Also, is there an equivalent rule for arrays?

**Comment by hzoo** [comment]
I think a `ignoreSingleLine` value would be ok for this then - similar to `requireTrailingComma` or `disallowSpaceAfterObjectKeys`.

**Comment by yaycmyk** [comment]
@hzoo Yup, that config would work nicely. Will update my request.

Doesn't look like that other PR has been merged yet?

**Comment by hzoo** [comment]
Yeah doesn't look like it. Although looking at it again - `requirePaddingNewLinesInObjects` is a different rule because it requires newlines for each curly brace (so nested objects). Here you are asking for newlines between the elements (I got confused) - which is similar to the current PR for arrays. So it seems like your previous request for an integer config makes more sense.

**Comment by TheSavior** [comment]
I'd like to see this as well. :+1:

**Comment by BridgeAR** [comment]
+1

**Comment by EladBezalel** [comment]
:+1:

**Comment by TheSavior** [comment]
It should probably take an integer to be consistent with http://jscs.info/rule/requireBlocksOnNewline.html

I would also only want this to be for basic types. Primitives or variables. For example, if I set the number to 1, I would not want this to pass:

```
var a = { foo: function() { return { obj: 'foo' } } };
```

**Comment by aronwoost** [comment]
+1

**Comment by markelog** [comment]
@aronwoost would you like to do it?

**Comment by aronwoost** [comment]
I'll at least try to provide tests.

**Comment by aronwoost** [comment]
@markelog Would it work with `"ignoreSingleValue"`? Or should we better have `maximum` and `ignoreBrackets` options like in [validateNewlineAfterArrayElements](http://jscs.info/rule/validateNewlineAfterArrayElements)? (see @hzoo above)

**Comment by markelog** [comment]
I think `ignoreSingleValue` is more suitable for this rule.

**Comment by hzoo** [comment]
We released [3.0](https://github.com/jscs-dev/node-jscs/releases/tag/v3.0.0) recently! We also wrote a [post](https://medium.com/@markelog/jscs-end-of-the-line-bc9bf0b3fdb2#.oot3wvgpu) about how 3.0 will be our last major release (since we are deprecating the project) and that we are [merging with ESLint](http://eslint.org/blog/2016/04/welcoming-jscs-to-eslint)!

Thus we won't be working on any new features moving forward which includes new rules/options. We will continue to fix bugs and work on CST/better autofixing for current rules in the upcoming months both to help JSCS users and to gain experience that we can hopefully use in ESLint. But otherwise we both teams will be working on getting [ESLint feature parity with JSCS](https://github.com/eslint/eslint/milestones/JSCS%20Compatibility) and just a nice transition for our users.

I'm putting an `orphaned` label to note this issue isn't planned to be worked on, but might be a good contribution as a custom rule (not in JSCS core itself).

I would encourage anyone who wants the rule to try writing it themselves: I would either write a custom rule or submit an issue to ESLint and if accepted a PR. Otherwise a custom rule that won't be in core repo itself.

JSCS rules are also very similar to ESLint rules: here are some resources to get started: 

- [Project-specific Lint Rules with ESLint](https://gist.github.com/jareware/7179093)
- [Creating Custom Rules for ESLint](https://insideops.wordpress.com/2015/12/08/creating-custom-rules-for-eslint)
- [Maintaining JavaScript Code Quality with ESLint ](https://www.paypal-engineering.com/2014/12/12/maintaining-javascript-code-quality-with-eslint/)
